records/2022-07-20.security.prefix-api-token.md

```diff
@@ -0,0 +1,39 @@
+# TITLE
+
+Date: 2022-07-20
+
+## Status
+
+**DRAFT**
+
+## Context
+
+The mishandling of application secrets has lead to many security incidents in companies big and small. Given the inherent risk in how users store these secrets, it is important that we provide mechanisms to protect our users from doing so improperly.  Taking a cue from [Github](https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/), [Slack](https://api.slack.com/authentication/token-types), and [Stripe](https://stripe.com/docs/api/authentication), we see that token prefixes are an effective technique to help us ensure our tokens don't end up in public repositories.
+
+By combining a known prefix with Github's [secret scanning program](https://docs.github.com/en/developers/overview/secret-scanning-partner-program), we can be alerted if one of our users mistakenly stores their API keys in the open and we can take action.
+
+## Options
+
+- Add a known prefix to our API keys in order to be able to easily identify them
+- Register with Github's secret scanning program and create an endpoint for them to report issues back to us
+- (optional) Give users a grace period to generate new keys
+  - After grace period, enforce a rule that ensures all API keys contain our prefix
```

Comment:
With GoC release cycles being what they are and the fact that some groups won't be able to update outside of them I would say this should not be an optional step.

Reply:
Because it will be a process that will take 1-3 months (hopefully no more but I wouldn't be surprised), with repeated warnings / education messages sent to the users, I agree we should remove the optional wording in there.

Reply:
Makes sense; I've updated the wording here.